crates/shared/src/ethrpc.rs

```diff
@@ -8,21 +8,79 @@ use self::{buffered::BufferedTransport, http::HttpTransport};
 use crate::http_client::HttpClientFactory;
 use ethcontract::{batch::CallBatch, dyns::DynWeb3, transport::DynTransport};
 use reqwest::{Client, Url};
-use std::convert::TryInto as _;
+use std::{
+    fmt::{self, Display, Formatter},
+    num::NonZeroUsize,
+};
 
 pub const MAX_BATCH_SIZE: usize = 100;
 
 pub type Web3 = DynWeb3;
 pub type Web3Transport = DynTransport;
 pub type Web3CallBatch = CallBatch<Web3Transport>;
 
+/// Command line arguments for the common Ethereum RPC connections.
+#[derive(clap::Parser)]
+#[group(skip)]
+pub struct Arguments {
```

Comment:
This isn't something you can fix here as it deserves a more general discussion, but I thought I'd mention it regardless. I don't think that this convention of sprinkling `claps` arguments along with logic unrelated to the CLI is the best. It's similar to how we use API objects in domain logic, but less severe since we don't seem to mind breaking our CLI interface. In this case though it's actually not that bad since `buffered_configuration` is called immediately and the code works with that instead. But I think it'd be better if `ethrpc` just re-exported `buffered::Configuration as BufferedConfiguration` and the module worked with that type only.

But anyway this isn't a huge problem just thought it's worth mentioning.

Reply:
> I don't think that this convention of sprinkling claps arguments along with logic unrelated to the CLI is the best.

So my reasoning here was that I wanted close locality between the CLI configuration and the component initialization logic. Note that we only include the `Arguments` in the `ethrpc` module, beside the `web3` function that is responsible for instantiating a `Web3` component for our processes and shouldn't be coupled with the actual component implementations themselves (specifically, the interface to the `buffered` module including the `BufferedTransport` is only the `Configuration` object, so we don't "leak" command line configuration abstraction in there).

Also, in the past I've found that command line configuration and component initialization is inherently coupled, so I find it convenient that changing how we initialize a component (either adding new options, changing semantics of existing flags, etc.) only affects one module instead of affecting multiple places.

I agree that this is not 100% ideal though (I would actually prefer "structured configuration" with `serde::Deserialize` and configuration files), but that is a much larger change.

That being said, I'm open to suggestions on better places to include these arguments.